subprojects/core/src/integTest/groovy/org/gradle/execution/taskgraph/ParallelTaskExecutionIntegrationTest.groovy

```diff
@@ -454,4 +463,40 @@ class ParallelTaskExecutionIntegrationTest extends AbstractIntegrationSpec {
             failure.assertHasCause "BOOM!"
         }
     }
+
+    def "other tasks are not started when an invalid task task is running"() {
+        given:
+        withParallelThreads(2)
+
+        expect:
+        2.times {
+            executer.expectDocumentedDeprecationWarning("Property 'invalidInput' has @Input annotation used on property of type 'File'. " +
+                "This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. " +
+                "Execution optimizations are disabled due to the failed validation. " +
+                "See https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.")
+            executer.withFullDeprecationStackTraceDisabled()
+
+            blockingServer.expect(":aInvalidPing")
+            blockingServer.expectConcurrent(":bPing", ":cPing")
+            run ":aInvalidPing", ":bPing", ":cPing"
+        }
+    }
+
+    def "invalid task is not executed in parallel with other task"() {
+        given:
+        withParallelThreads(2)
```

Comment:
```suggestion
        withParallelThreads(3)
```

This needs to be at least `3`, since if it is only `2` the test won't fail since the two parallel execution slots are taken up by `aPing` and `bPing` and we'll never try to schedule `cInvalidPing` anyway.

Reply:
Ah, yes, I originally only had two tasks, and forgot to increase this when adding the third.